src/domains/preference/preference.domain.ts

```diff
@@ -0,0 +1,61 @@
+import { AccessTokenDomain } from '../auth/access-token.domain'
+import { IPreference } from './index.interface'
+import { PreferenceModel, PreferenceProps } from '@/schemas/preference.schema'
+import { BadRequestError } from '@/errors/400/index.error'
+import { GlobalLanguageCode } from '@/global.interface'
+
+export class PreferenceDomain {
+  private readonly props: Partial<IPreference>
+
+  private constructor(props: Partial<IPreference>) {
+    this.props = props
+  }
+
+  toResDTO(): Partial<IPreference> {
+    return this.props
+  }
+
+  static async fromMdb(
+    atd: AccessTokenDomain,
+    model: PreferenceModel,
+    avoidRecursiveCall = false,
+  ): Promise<PreferenceDomain> {
+    const preferenceDocs = await model.find({ ownerID: atd.userId }).exec()
+    if (avoidRecursiveCall && preferenceDocs.length !== 1) {
+      throw new BadRequestError(
+        'Something went really wrong while creating preference',
+      )
+    }
```

Comment:
I think that it can be deleted?
1. Make a PreferenceDomain from slicedDocs of `preferenceDocs.slice(1)`
2. Loop through each and do the `PreferenceDomain.delete()`
3. Check if this logic can be applied to `SupportDomain`

Reply:
delete preference should be always private with comment
User must have at least one preference data, and therefore should not call?
Or if you reset it, then sure you may delete it.
But since such feature does not exist, it is considered private

Write the information above

Reply:
Handled in https://github.com/ajktown/api/pull/58/commits/4c7b2b0c20495784227499800b6d8c75e38c59b9